Amazon.QLDB.Driver/session/BaseQldbSession.cs

```diff
@@ -58,5 +48,10 @@ internal string GetSessionId()
         {
             return this.session.SessionId;
         }
+
+        private protected static bool IsTransactionExpiredException(InvalidSessionException ise)
+        {
+            return Regex.Match(ise.Message, @"Transaction\s.*\shas\sexpired").Success;
```

Comment:
What is the transaction ID, is it a base64 encoded ID, or a numeric ID, or a hex ID? Right now .* is too permissive, we should restrict this to what can actually appear in the message, or at least make the regex non-greedy. Just to double check as well that a ticket has been created to remove this nasty bit of tech debt?

Reply:
I believe it to be alphanumeric but don't have the finer details. The current implementation was done by Seattle-side and I'm not sure there's much value in making this less greedy, and that would also imply making a change to all drivers.

The ticket hasn't been made yet as since talking about this last we didn't have a "normal" stand-up with Qiusheng or Ian. I plan to raise it tomorrow. We should push on that though instead of trying to improve something that's a stopgap.